### DISCUSSION TITLE
How to use REST API to set user profile status?

### DISCUSSION THREAD

**Discussion by UlyssesZh** [community/community#108473] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

We can [set profile status](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/customizing-your-profile/personalizing-your-profile#setting-a-status). How to do this using REST API? I cannot find it in the API documentation.

**Comment by bewuethr** [comment, marked as answer]
You can't do it in the REST API, but via [GraphQL](https://docs.github.com/en/graphql/reference/mutations#changeuserstatus). Using the GitHub CLI, you could do something like this:

```sh
emoji=':juggling_person:'
status='This is my status'

gh api graphql \
    --raw-field emoji="$emoji" \
    --raw-field status="$status" \
    --raw-field query='
        mutation UpdateStatus($emoji: String!, $status: String!) {
          changeUserStatus(input:{emoji: $emoji, message: $status}) {
            clientMutationId
            status {
              emoji
              message
              updatedAt
              user {
                login
              }
            }
          }
        }
    '
```

There's also an extension to the GitHub CLI for specifically this: [gh-user-status](https://github.com/vilmibm/gh-user-status). You can install and use it like this:

```sh
gh ext install vilmibm/gh-user-status
gh user-status set 'This is my status' --emoji='juggling_person'
```

Be careful about the placement of colons for the emoji string: they're required for the direct GraphQL call, but not required when using the user-status extension.

**Reply by UlyssesZh** [reply]
Thank you! How to clear the status?
Never mind. I can set the expiry time to one second.
